test/index.test.js

```diff
@@ -21,311 +19,307 @@ function objectIdTest (t, ObjectId, message) {
   message = message || 'expect ObjectId value'
 
   const obj1 = new ObjectId()
-  t.ok(obj1)
+  t.assert.ok(obj1)
 
   const obj2 = new ObjectId(obj1)
-  t.ok(obj2)
-  t.ok(obj1.equals(obj2))
+  t.assert.ok(obj2)
+  t.assert.ok(obj1.equals(obj2))
 
   const obj3 = new ObjectId()
-  t.ok(!obj1.equals(obj3))
+  t.assert.ok(!obj1.equals(obj3))
 }
 
-function clientTest (t, client, message) {
+async function clientTest (t, client, message) {
   t.plan(1)
   message = message || 'expect client'
   const db = client.db(DATABASE_NAME)
 
   const col = db.collection(COLLECTION_NAME)
 
-  t.resolves(async () => {
-    const r = await col.insertMany([{ a: 1 }])
-    assert.strictEqual(1, r.insertedCount)
-  }, message)
+  const r = await col.insertMany([{ a: 1 }])
+  t.assert.strictEqual(1, r.insertedCount)
 }
 
-function databaseTest (t, db, message) {
+async function databaseTest (t, db, message) {
   t.plan(1)
   message = message || 'expect database'
 
   const col = db.collection(COLLECTION_NAME)
 
-  t.resolves(async () => {
-    const r = await col.insertMany([{ a: 1 }])
-    assert.deepEqual(1, r.insertedCount)
-  }, message)
+  const r = await col.insertMany([{ a: 1 }])
+  t.assert.strictEqual(1, r.insertedCount)
 }
 
 test('re-export ObjectId', async (t) => {
   t.plan(1)
-  t.test(async t => objectIdTest(t, fastifyMongo.ObjectId))
+  await t.test(async t => objectIdTest(t, fastifyMongo.ObjectId))
 })
 
 test('re-export ObjectId destructured', async (t) => {
   t.plan(1)
-  t.test(async t => objectIdTest(t, ObjectId))
+  await t.test(async t => objectIdTest(t, ObjectId))
 })
 
 test('export of mongodb', async (t) => {
   t.plan(2)
-  t.same(typeof fastifyMongo.mongodb.ObjectId, 'function')
-  t.same(fastifyMongo.mongodb.BSONType.array, 4)
+  t.assert.strictEqual(typeof fastifyMongo.mongodb.ObjectId, 'function')
+  t.assert.strictEqual(fastifyMongo.mongodb.BSONType.array, 4)
 })
 
 test('{ url: NO_DATABASE_MONGODB_URL }', async (t) => {
   t.plan(6)
 
   const fastify = await register(t, { url: NO_DATABASE_MONGODB_URL })
 
-  t.ok(fastify.mongo)
-  t.ok(fastify.mongo.client)
-  t.ok(fastify.mongo.ObjectId)
-  t.notOk(fastify.mongo.db)
+  t.assert.ok(fastify.mongo)
+  t.assert.ok(fastify.mongo.client)
+  t.assert.ok(fastify.mongo.ObjectId)
+  t.assert.ok(!fastify.mongo.db)
```

Comment:
IMHO `ifError` or `strictEqual` reads better here than looking for `!` [ref](https://nodejs.org/api/assert.html#assertiferrorvalue)
```suggestion
  t.assert.ifError(fastify.mongo.db)
```
or
```js
  t.assert.strictEqual(fastify.mongo.db, undefined)
```